catkin_ws/src/state_estimation/src/state_aggregator.py

```diff
@@ -7,65 +7,75 @@
 from tf2_ros import TransformBroadcaster
 
 from geometry_msgs.msg import Pose, Quaternion, Vector3, TransformStamped, Point
-from sensors import DepthSensor, IMU, DVL
-from std_msgs.msg import Float64, Bool
+from sensors import DepthSensor, IMU, IMUFrontCamera, DVL
+from std_msgs.msg import Float64, Int32
 from rosgraph_msgs.msg import Clock
 
 DEG_PER_RAD = 180 / np.pi
 
+def swap_main_sensor_message(axis_index, sensor_index, sensor_name):
+    global last_warning_message_time
+
+    if sensor_index != 0 and rospy.get_time() - last_warning_message_time[axis_index] > sensor_swap_warning_interval:
+            last_warning_message_time[axis_index] = rospy.get_time()
+            rospy.logwarn("Using {} for {}.".format(sensor_name, axis_names[axis_index]))
+
```

Comment:
this function is responsible for printing the warning message when the "main" (most important) sensor for a specific axis stops working and switches to another - not related to status check.

To avoid spamming, only print warning if the current sensor is not the main sensor (index_sensor is the index of the sensor in the sensor_priorities dictionary below, so index = 0 is the main sensor) AND it has been at least 5 seconds since the last warning